components/main_menu/main_menu.tsx

```diff
@@ -475,37 +472,29 @@ export class MainMenu extends React.PureComponent<Props> {
-                                                </span>
-                                                <span className='message'>
-                                                    {this.props.isFreeTrial ? (
-                                                        formatMessage({id: 'navbar_dropdown.create.tooltip.cloudFreeTrial', defaultMessage: 'During your trial you are able to create multiple teams. These teams will be archived after your trial.'})
-                                                    ) : (
-                                                        formatMessage({id: 'navbar_dropdown.create.tooltip.cloudFree', defaultMessage: 'This is a paid feature, available with a free 30-day trial'})
-                                                    )}
-                                                </span>
-                                            </Tooltip>
-                                        )}
-                                    >
-                                        {this.props.isFreeTrial ? (
-                                            <i className='MenuItem__icon-tooltip icon trial'/>
-                                        ) : (
-                                            <ToggleModalButton
-                                                modalId={ModalIdentifiers.CREATE_TEAM_RESTRICTED_MODAL}
-                                                dialogType={CreateTeamRestrictedModal}
-                                            >
-                                                <i className='MenuItem__icon-tooltip icon icon-key-variant'/>
-                                            </ToggleModalButton>
-                                        )}
-                                    </OverlayTrigger>
-                                </span>
+                                <RestrictedIndicator
+                                    blocked={!this.props.isFreeTrial}
+                                    tooltipMessage={formatMessage({
+                                        id: 'navbar_dropdown.create.tooltip.cloudFreeTrial',
+                                        defaultMessage: 'During your trial you are able to create multiple teams. These teams will be archived after your trial.',
+                                    })}
+                                    modalTitle={formatMessage({
+                                        id: 'navbar_dropdown.create.modal.title',
+                                        defaultMessage: 'Try unlimited teams with a free trial',
+                                    })}
+                                    modalMessage={formatMessage({
+                                        id: 'navbar_dropdown.create.modal.description',
+                                        defaultMessage: 'Create unlimited teams with one of our paid plans. Get the full experience of Enterprise when you start a free, 30 day trial.',
```

Comment:
Nit, can we use the [FREEMIUM_TO_ENTERPRISE_TRIAL_LENGTH_DAYS](https://github.com/mattermost/mattermost-webapp/blob/master/utils/cloud_utils.ts#L36) constant and pass it in as a value for this text? That will help reduce translation churn going forward, and make any changes to trial length easier.